README.md

```diff
@@ -8,4 +8,245 @@ are in the process of being changed and will certainly break your code.
 [travis-image]: https://travis-ci.org/census-instrumentation/opencensus-go.svg?branch=master
 [travis-url]: https://travis-ci.org/census-instrumentation/opencensus-go
 [godoc-image]: https://godoc.org/github.com/census-instrumentation/opencensus-go?status.svg
-[godoc-url]: https://godoc.org/github.com/census-instrumentation/opencensus-go
\ No newline at end of file
+[godoc-url]: https://godoc.org/github.com/census-instrumentation/opencensus-go
+
+# Go stats core library
+
+## Keys and tagging
+
+### To create/retrieve a key
+A key is defined by its name. To use a key a user needs to know its name and type (currently only keys of type string are supported. Later support for keys of type int64 and bool will be supported). Calling CreateKeyString(...) multiple times with the same name returns the same key.
+To create/retrieve a key the user calls:
+
+Create/retrieve key:
+
+    if err, key1 := tags.CreateKeyString("keyNameID1"); err != nil {
+        // handle error
+    }
+    ...
+
+    if err, key2 := tags.CreateKeyString("keyNameID2"); err != nil {
+        // handle error
+    }
+    ...
+
+### Create a set of tags associated with keys
+To create a new tag set from scratch using changes:
+
+    tsb := &tags.TagSetBuilder{}
+    tsb..StartFromEmpty()
```

Comment:
tsb.StartFromEmpty()

StartFromX pattern is a very uncommon pattern in Go. Users expect either to thing work with a zero value or a constructor is present.

Couldn't this package just export:
```
func NewFromTagSet(ts *TagSet) *TagSetBuilder
func (ts *TagSet) Empty() *TagSetBuilder
```

Reply:
Replaced it with:
func EmptyTagSetBuilder() TagSetBuilder
func NewTagSetBuilderFromTagSet(ts *TagSet) TagSetBuilder

where TagSetBuilder is an interface now.

Reply:
I completely agree that we don't literally need an EmptyTagSetBuilder. 

In regard to the Builder pattern I think we would like to have TagSet immutable to simplify and optimize the synchronization schema, and then another possible option would be to have Mutation and have a NewTagSet(Mutation ...) where a mutation is one of the 3.

Reply:
Replaced it with a single NewTagSetBuilder(ts *TagSet) method.

To hyangah@ tte builder pattern is for efficiency purposes. Once the TagSet is built we want it to be immutable. To avoid making copies when adding 10 or 100 KeyVals I think a builder pattern is appropriate here.